src/engines/kotor/gui/options/resolution.cpp

```diff
@@ -22,23 +22,64 @@
  *  The graphics resolution menu.
  */
 
+#include <SDL_video.h>
+#include "src/common/strutil.h"
+#include "src/common/ustring.h"
+
+#include "src/graphics/windowman.h"
+
 #include "src/engines/kotor/gui/options/resolution.h"
 
 #include "src/engines/aurora/kotorjadegui/panel.h"
+#include "src/engines/aurora/kotorjadegui/listbox.h"
 
 namespace Engines {
 
 namespace KotOR {
 
-OptionsResolutionMenu::OptionsResolutionMenu(Console *console) : GUI(console) {
+OptionsResolutionMenu::OptionsResolutionMenu(Console *console) : GUI(console), _newWidth(0), _newHeight(0) {
 	load("optresolution");
 
 	WidgetPanel *guiPanel = getPanel("TGuiPanel");
 	guiPanel->setPosition(-guiPanel->getWidth()/2, -guiPanel->getHeight()/2, 0);
+
+	int currentIndex = -1;
+	std::vector<Graphics::DisplayMode> modes = WindowMan.getDisplayModes();
```

Comment:
This only accounts for resolutions the display says it can natively do, so only, I guess, fullscreen modes the display advertises. This ignores wanting to run it in windowed mode in a non-native resolution.

So I'd add more resolutions to that list. For example, the resolutions the original KotOR binary supports. And just a list of general common resolutions. But you might want only add resolutions the native display could fit, i.e. get the max width and height of all display modes, and only add resolutions <= that max width and height.